### ISSUE TITLE
LSL support?

### ISSUE THREAD

**Issue by behinger** [mindhiveproject/You-Quantified#1]
Hi!
I hope you are doing fine!

I think this was discussed at MOBI, and it was positively affirmed, will there be LSL support?

Edit: Ok, I have now found some code indicating some support, but it is not available via the website. any "tricks"?

Cheers, Bene

PS: Reminder for the license ;)

**Comment by esromerog** [comment]
Hi! 

Currently, LSL support was added with rather makeshift solution. If it gets many requests or a new API becomes available that may allow me to easily access LSL streams (TCP/UDP connections) from the web browser, I might look into a better alternative.

**Comment by behinger** [comment]
Hi! Thanks for the answer.

I didnt see LSL at all when using - so it is only available with some tricks? 

I further don't really understand anything from the more technical site as I'm not into browser-programming things :S

**Comment by behinger** [comment]
Bump on this - any update? I have a group of students starting on a demo, I wonder if I can use this or we develop our own visualization tool. 

Cheers, Bene

**Comment by esromerog** [comment]
Hi!

Sorry for not getting back to this before! LSL was a very niche use-case, so the current public version had some changes that removed support; however, the code is still there and would only require some tweaking. Hopefully I can get back to it really soon and officially re-add the support.

In the meantime, you can use the "old" version of the platform which does have LSL support, although it comes some extra steps:
1. Go into the [website](https://creative-quantified-self.netlify.app/devices) and add a new device. Hold the `alt` key while the new device dropdown menu is open to select LSL.
2. On the computer where you are hosting the LSL stream, run the Python script that can be found on [this repository](https://github.com/esromerog/LSLWebsocketMirror) by following the instructions.

I hope to provide a more seamless solution in the very near future! I'll keep this thread updated once that comes out. Sorry to address this so late and hope that solution works well for your demo.

Best,
Esteban

**Comment by esromerog** [comment]
Hi, I'm glad to announce that LSL support has been added to the app's current release version! I have also updated the [Python repository](https://github.com/esromerog/LSLWebsocketMirror) with some minor improvements.

LSL support includes some nice features such as:
- Saving data with the original timestamps and YQ timestamps
- Being able to record data in your local LSL Lab Recorder session and compare it to YQ

The platform is mainly for real-time visualization. Any recordings will be made based on the stream with the highest sampling rate and may not be accurate.

If you run into any issues feel free to open a new thread, for now, I'm closing this issue.
